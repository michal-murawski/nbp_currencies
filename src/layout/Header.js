import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import AppBar from 'material-ui/AppBar';
import { HeaderStyles } from './styles';

export const headerTitle = 'Wybierz swoje ulubione waluty';

const Header = props => {
  const { classes, children } = props;

  return (
    <AppBar position="static" color="default">
      <Typography variant="display3" className={classes.title}>
        Wybierz swoje ulubione waluty
      </Typography>
      {children}
    </AppBar>
  );
};

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export const HeaderRaw = Header;

export default withStyles(HeaderStyles)(Header);
