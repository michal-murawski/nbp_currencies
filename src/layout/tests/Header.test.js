import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Typography from 'material-ui/Typography';
import AppBar from 'material-ui/AppBar';
import { HeaderRaw, headerTitle } from '../Header';

const TestChild = () => <div>Test child text</div>;
const mockClasses = { title: 'title_class' };

describe('layout/Header', () => {
  const wrapper = shallow(
    <HeaderRaw classes={mockClasses}>
      <TestChild />
    </HeaderRaw>
  );

  it('should render properly the header', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render AppBar and Typography', () => {
    expect(wrapper.find(AppBar)).toHaveLength(1);
    expect(wrapper.find(Typography)).toHaveLength(1);
    expect(
      wrapper
        .find(Typography)
        .render()
        .text()
    ).toEqual(headerTitle);
  });
});
