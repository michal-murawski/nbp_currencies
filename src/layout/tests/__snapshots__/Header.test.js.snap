// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout/Header should render properly the header 1`] = `
<WithStyles(AppBar)
  color="default"
  position="static"
>
  <WithStyles(Typography)
    className="title_class"
    variant="display3"
  >
    Wybierz swoje ulubione waluty
  </WithStyles(Typography)>
  <TestChild />
</WithStyles(AppBar)>
`;
